api/napi/src/interpreter/component_instance.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+use napi::{Env, Error, JsFunction, JsUnknown, NapiRaw, NapiValue, Ref, Result};
+use slint_interpreter::{ComponentHandle, ComponentInstance, Value};
+
+use super::JsComponentDefinition;
+
+#[napi(js_name = "JsComponentInstance")]
+pub struct JsComponentInstance {
+    inner: ComponentInstance,
+}
+
+impl From<ComponentInstance> for JsComponentInstance {
+    fn from(instance: ComponentInstance) -> Self {
+        Self { inner: instance }
+    }
+}
+
+#[napi]
+impl JsComponentInstance {
+    #[napi(constructor)]
+    pub fn new() -> Self {
+        unreachable!("ComponentDefinition can only be created by using ComponentCompiler.")
+    }
+
+    #[napi]
+    pub fn definition(&self) -> JsComponentDefinition {
+        self.inner.definition().into()
+    }
+
+    #[napi]
+    pub fn run(&self) {
+        self.inner.run().unwrap()
+    }
+
+    #[napi]
+    pub fn get_property(&self, env: Env, name: String) -> Result<JsUnknown> {
+        let value = self
+            .inner
+            .get_property(name.as_ref())
+            .map_err(|e| Error::from_reason(format!("{e}")))?;
+        super::value::to_js_unknown(&env, &value)
+    }
+
+    #[napi]
+    pub fn set_property(&self, env: Env, name: String, js_value: JsUnknown) -> Result<()> {
+        let value = super::value::to_value(
+            &env,
+            &self.inner,
+            js_value,
+            &self
+                .inner
+                .get_property(&name)
+                .map_err(|_| napi::Error::from_reason("Cannot get property."))?,
+        )?;
+        self.inner.set_property(&name, value).map_err(|e| Error::from_reason(format!("{e}")))?;
+        Ok(())
+    }
+
+    #[napi]
+    pub fn set_callback(&self, env: Env, name: String, callback: JsFunction) -> Result<()> {
+        let function_ref = RefCountedReference::new(&env, callback)?;
+        self.inner
+            .set_callback(name.as_str(), move |values| {
+                let callback: JsFunction = function_ref.get().unwrap();
+                let result = callback
+                    .call(
+                        None,
+                        values
+                            .iter()
+                            .map(|v| super::value::to_js_unknown(&env, v).unwrap())
+                            .collect::<Vec<JsUnknown>>()
+                            .as_ref(),
+                    )
+                    .unwrap();
+
+                super::js_unknown_to_value(env, result).unwrap()
```

Comment:
As mentioned above, this is where you need to use the return type of the callback (as `langtype::Type`). Basically `js_unknown_to_value` should not exist, I think, and it should always be `value::to_value`.